app/api/business/validators/application_validator.py

```diff
@@ -8,204 +9,384 @@ def __init__(self, application):
 
     def validate_all(self):
         return (self.validate_basics() +
+                self.validate_details() +
+                self.validate_contacts() +
+                self.validate_disclosures() +
                 self.validate_documents() +
+                self.validate_methods() +
+                self.validate_recruiter() +
+                self.validate_services() +
                 self.validate_pricing() +
+                self.validate_case_studies() +
                 self.validate_candidates() +
-                self.validate_case_studies())
+                self.validate_products())
 
-    def validate_basics(self):
+    def __validate_required(self, data, required_fields, step):
         errors = []
-        if not self.application.data.get('name'):
+        for field, label in required_fields.iteritems():
+            if not data.get(field):
+                errors.append({
+                    'field': field,
+                    'message': '{} is required'.format(label),
+                    'severity': 'error',
+                    'step': step
+                })
+        return errors
+
+    def validate_basics(self):
```

Comment:
I think it might be worthwhile validating the LinkedIn link if it's been provided.  Attempting to save with a bad link results in an error message that isn't clear.

Reply:
added http validation